test_wazuh/test_fim/test_checks/test_check_all.py

```diff
@@ -130,3 +131,40 @@ def test_check_all(path, checkers, get_configuration, configure_environment, res
 
     regular_file_cud(path, wazuh_log_monitor, min_timeout=15, options=checkers,
                      time_travel=get_configuration['metadata']['fim_mode'] == 'scheduled')
+
+
+@pytest.mark.parametrize('path, checkers', [(testdir1, {})])
+def test_check_all_no(path, checkers, get_configuration, configure_environment, restart_syscheckd, wait_for_initial_scan):
+    """Test the functionality of `check_all` option when set to no.
+
+    When setting `check_all` to no, only 'type' and 'checksum' attributes should appear in every event. This will
+    avoid any modification event.
+
+    This test is intended to be used with valid configurations files. Each execution of this test will configure the
+    environment properly, restart the service and wait for the initial scan.
+
+    Parameters
+    ----------
+    path : str
+        Directory where the file is being created and monitored
+    checkers : dict
+        Dict with all the check options to be used
+    """
+    check_apply_test({'test_check_all_no'}, get_configuration['tags'])
+    scheduled = get_configuration['metadata']['fim_mode'] == 'scheduled'
+
+    # Create regular file and dont expect any check
+    file = 'regular'
+    create_file(REGULAR, path, file)
+    check_time_travel(scheduled)
+    create_event = wazuh_log_monitor.start(callback=callback_detect_event, timeout=15).result()
+    assert create_event['data']['type'] == 'added'
+    assert list(create_event['data']['attributes'].keys()) == ['type', 'checksum']
+
+    # Create regular file and dont expect any check. Since it is not using any check, modification events will not
+    # be triggered
+    delete_file(path, file)
+    check_time_travel(scheduled)
```

Comment:
We need to decide if `check_all="no"` should generate events or no when creating or deleting. Furthermore, the test must check if changing any attribute or content in any file does not generate any event.

Reply:
It is confirmed that when creating or deleting a file with `check_all="no"`, an event should be generated with only the following attributes: `type`, `checksum`.

No events should be generated when modifying said file.